engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -171,7 +172,7 @@ public ComponentSystem get(String name) {
         return namedLookup.get(name);
     }
 
-    public Iterable<ComponentSystem> iterateAll() {
+    public List<ComponentSystem> iterateAll() {
```

Comment:
What's the reason to make this more specific? I'm not super familiar with the Java best practices, but `iterateAll()` a) sounds like it would **do something** instead of return something, and b) all we can say about it is that it is iterable…

A better name might be `public Iterable<ComponentSystem> getAllComponentSystems()`(?)

Reply:
IMO, when defining an interface, you should use whatever return time fits the best, but when implementing actual code, you should always use the most specific class you can, unless there is a real reason not to. Better name might be `getAllSystems` though.

Reply:
Let's change it (if we worry about changing API we should rename and leave a `@deprecated` version of the old name in place…).